iib/web/models.py

```diff
@@ -1075,6 +1095,170 @@ def get_mutable_keys(self):
         return rv
 
 
+class RequestMergeIndexImage(Request):
+    """A "merge-index-image" build request."""
+
+    __tablename__ = 'request_merge_index_image'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+    binary_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    binary_image_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    binary_image = db.relationship('Image', foreign_keys=[binary_image_id], uselist=False)
+    binary_image_resolved = db.relationship(
+        'Image', foreign_keys=[binary_image_resolved_id], uselist=False
+    )
+
+    deprecation_list = db.relationship('Image', secondary=BundleDeprecation.__table__)
+
+    index_image_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    index_image = db.relationship('Image', foreign_keys=[index_image_id], uselist=False)
+
+    source_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    source_from_index_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    source_from_index = db.relationship('Image', foreign_keys=[source_from_index_id], uselist=False)
+    source_from_index_resolved = db.relationship(
+        'Image', foreign_keys=[source_from_index_resolved_id], uselist=False
+    )
+
+    target_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    target_index_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    target_index = db.relationship('Image', foreign_keys=[target_index_id], uselist=False)
+    target_index_resolved = db.relationship(
+        'Image', foreign_keys=[target_index_resolved_id], uselist=False
+    )
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['merge_index_image'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the merge-index-image API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        deprecation_list = request_kwargs.pop('deprecation_list', [])
+        if not isinstance(deprecation_list, list) or any(
+            not item or not isinstance(item, str) for item in deprecation_list
+        ):
+            raise ValidationError(
+                '"deprecation_list" should be an empty array or an array of strings.'
```

Comment:
To match the rest of the errors
```suggestion
                '"deprecation_list" should be an empty array or an array of strings'
```